### PR TITLE
Type hint complete Albert model file.

### PR DESCRIPTION
**Pull request by karthikrangasai**
# What does this PR do?
Type hint the Albert Model file for PyTorch Model.

Part of #16059 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@Rocketknight1

### CODE REVIEW

**Review by kumar-abhishek — APPROVED** [review]
Looks good to me.

**Comment by Rocketknight1** [comment]
Hi, I'm sorry for the delay with this one! We really appreciate it, and I'm trying to get a chance to look through it all ASAP!

**Review by Rocketknight1 — APPROVED** [review]
This is great, thank you! The return type annotations are incredibly complete - I'm going to check with the team if that's okay; it might become very difficult to maintain that if we ever change the function signature.

It'll pain me to see it go if we do decide to just use `Tuple`, though, because it was a thing of beauty.

**Comment by karthikrangasai** [comment]
Thanks a lot and I am glad that you liked the work.
More of these will be coming in a few days for other models, so it would be great if I know up to what level of detailing is needed for the Type Annotations.

**Comment by Rocketknight1** [comment]
@karthikrangasai I spoke to the team and the conclusion was that we should just use `Union[AlbertForPreTrainingOutput, Tuple]` - as well as being easier for you, when these type hints are copied into the documentation, it'll be a lot more readable that way.

That said, I respect the dedication and precision that went into making it, so I'm a little sad to see it go.

**Comment by karthikrangasai** [comment]
Hello @Rocketknight1 ,

Sure, no worries.
I can make the changes and update the Pull Request in a while.


Although I am not sure what the reason is for to have return type a Tuple, a suggestion is to maybe remove the return type as  "Tuple" and make it the respective output type for all models. This will have keyword based output values like 
```
output = AlbertModel(**inputs)

output.last_hidden_state
output.pooler_output
```
and this might become a more cleaner API.

File: src/transformers/models/albert/modeling_albert.py [hunk 1]
```diff
@@ -392,22 +408,31 @@ def forward(
...
+        hidden_states: torch.Tensor,
+        attention_mask: Optional[torch.FloatTensor] = None,
+        head_mask: Optional[torch.FloatTensor] = None,
+        output_attentions: bool = False,
+        output_hidden_states: bool = False,
+    ) -> Union[
+        Tuple[torch.Tensor],
+        Tuple[torch.Tensor, Tuple[torch.Tensor, ...]],
+        Tuple[torch.Tensor, Tuple[torch.Tensor, ...], Tuple[torch.Tensor, ...]],
+    ]:
```
Comment (Rocketknight1) [inline src/transformers/models/albert/modeling_albert.py:435]:
```suggestion
    ) -> Tuple[Union[torch.Tensor, Tuple[torch.Tensor]], ...]:
```
Simplifying this one a bit to save space!

**Review by Rocketknight1 — APPROVED** [review]
Overall this looks great, thank you! And sorry for the slow reply. I made one change, but other than that we should be good to go!

**Comment by Rocketknight1** [comment]
Seeing some code quality issues, I'm guessing because we changed our versions for code formatting tools and might need to rebase. Let me check!

**Comment by Rocketknight1** [comment]
It seems like you'll need to pull commits from our repo to the main branch of your repo, then rebase your branch, then force push to update. After that, the files should be updated and the error should go away!

**Comment by karthikrangasai** [comment]
Hello @Rocketknight1, I have updated the PR with the main branch.
All tests are passing.

**Comment by Rocketknight1** [comment]
Great job. Thank you for this PR, it's much appreciated!

**Pull request merged**